code/modules/clothing/glasses/scanners.dm

```diff
@@ -158,6 +161,41 @@ var/list/meson_wearers = list()
 		visible_message("<span class = 'warning'>\The [src] sputter out.</span>")
 		disable()
 
+/obj/item/clothing/glasses/scanner/meson/proc/clear()
+	if (viewing)
+		meson_wearers -= viewing
+		if (viewing.client)
+			viewing.client.images -= false_wall_images
+
+/obj/item/clothing/glasses/scanner/meson/proc/apply()
+	if (!viewing || !viewing.client || !on)
+		return
+
+	meson_wearers += viewing
+	viewing.client.images += false_wall_images
+
+/obj/item/clothing/glasses/scanner/meson/unequipped(var/mob/M)
+	update_mob()
+	..()
+
+/obj/item/clothing/glasses/scanner/meson/equipped(var/mob/M)
+	update_mob(M)
+	..()
+
+/obj/item/clothing/glasses/scanner/meson/proc/update_mob(var/mob/new_mob)
+	if (new_mob == viewing)
+		clear()
+		apply()
+		return
+
+	if (new_mob != viewing)
```

Comment:
isn't this always true?

Reply:
Nope. The proc is called with a null arg to clear() and remove the viewer. Same as with the material scanner below.

Reply:
The `if` before this one checks whether `new_mob` is equal to `viewing`. If it is, it returns. So the only way to reach this second if, is for `new_mob` to not be equal to `viewing`.

Reply:
oh, well then I fix it in both places